sequencing/ATACSeq/jobSubmission/3_batchRunPeakCalling.sh

```diff
@@ -6,107 +6,158 @@
+## $1 -> <config file> directory to config file for the corresponding project                                         ||
+## $2 -> <option> Specify step to run: SHIFT, PEAKS, FRIP, HMMRATAC. Can be combined. Default is to run all           ||
+##                                                                                                                    ||
+## DESCRIPTION: This script performs the core analysis of the ATAC-seq pipeline, which is calling peaks at sample     ||
+## level. The three modes available in MACS3 are used for this in order to compare results.                           ||
+##                                                                                                                    ||
+## ===================================================================================================================##
+
+## ============ ##
+##    SET UP    ##
+## ============ ##
 
 ## print start date and time
 echo Job started on:
 date -u
-	
-## needs to be executed from the scripts folder
-echo "Changing Folder to: "
-echo $SLURM_SUBMIT_DIR
 
-cd $SLURM_SUBMIT_DIR
-
-## load config file provided on command line when submitting job
+## load config file provided on command line related to the specified project
+source $1
 echo "Loading config file for project: " $1
-export PROJECT=$1
-
-source ./ATACSeq/config/config.txt 
 echo "Project directory is: " $DATADIR
 
+## Log files directory
+LOG_DIR=ATACSeq/logFiles/${USER}/${SLURM_ARRAY_JOB_ID}
+echo "Log files will be moved to dir: " $LOG_DIR
+mkdir -p $LOG_DIR
+mv ATACPeakCallingS3-${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}* $LOG_DIR
+
 ##check array specified and exit if not
 if [[ ${SLURM_ARRAY_TASK_ID} == '' ]]
 then 
     { echo "Job does not appear to be an array. Please specify --array on the command line." ; exit 1; }
 fi
 
-#-----------------------------------------------------------------------#
-
-echo "Changing Folder to Data directory "
-echo ${ALIGNEDDIR}
+## check step method matches required options and exit if not
+if [[ ! $2 =~ "SHIFT" ]] && [[ ! $2 =~ "PEAKS" ]] && [[ ! $2 =~ "HMMRATAC" ]] && [[ ! $2 =~ "FRIP" ]] &&[[ ! $2 == '' ]];
+then 
+    { echo "Unknown step specified. Please use SHIFT, PEAKS, HMMRATAC, FRIP or some combination of this as a single string (i.e. FASTQC,TRIM)" ; exit 1; }            
+fi
 
 cd ${ALIGNEDDIR}
 BAMFILES=($(ls *.filt.nodup.bam))
-
-echo "Number of bam files found for alignment:"" ""${#BAMFILES[@]}"""	
-
-sample=${BAMFILES[${SLURM_ARRAY_TASK_ID}]}
-sampleName=$(basename ${sample%.filt.nodup.bam})
-
-
+echo "Number of aligned bam files: "" ""${#BAMFILES[@]}"""
+  
+## ============ ##
+##    STEPS     ##
+## ============ ##
+
+## If only $1 is specified, all steps will be run
+## option SHIFT: shift aligned reads for single-end peak calling   
 if [ $# = 1 ] || [[ $2 =~ 'SHIFT' ]]
 then
 	module load BEDTools
 	module load SAMtools
```

Comment:
I don't think these modules are compatible, they use different versions of GCC. With BEDtools using GCC 9.3.0 and SAMtools using GCC 7.3.0.

You could specify:
SAMtools/1.11-GCC-9.3.0
BEDTools/2.29.2-GCC-9.3.0

hopefully the script still works with the older version of SAMtools.

If it doesn't you could also use:
BEDTools/2.27.1-foss-2018b
SAMtools/1.9-foss-2018b